distribution/pull_v2.go

```diff
@@ -606,13 +620,29 @@ func (p *puller) pullSchema1(ctx context.Context, ref reference.Reference, unver
 	return imageID, manifestDigest, nil
 }
 
+func checkSupportedMediaType(mt string) error {
+	mt = strings.ToLower(mt)
+	isOCIImageType := strings.HasPrefix(mt, "application/vnd.oci.image")
+	isDockerType := strings.HasPrefix(mt, "application/vnd.docker.")
+
+	if !isOCIImageType && !isDockerType {
+		return unsupportedMediaTypeError{MediaType: mt}
```

Comment:
Nit: it could be misleading to report the transformed media-type in the error. Parameters may follow the type and subtype, [and they could be case-sensitive.](https://httpwg.org/specs/rfc9110.html#field.content-type:~:text=media%20type%20registry.-,Parameter%20values%20might%20or%20might%20not%20be%20case%2Dsensitive%2C%20depending%20on%20the%20semantics%20of%20the%20parameter%20name.,-For%20example%2C%20the)

Reply:
Thanks, fixed!